### PR TITLE
Clean up `php 8.0`.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ❌

- [x] Add union types.
- [x] Fix phpdoc.
- [x] Add use function. 
- [x] Upload code coverage 100%.
- [x] Upload infection 99,99%.
- [x] Clean up code.
- [x] Add more tests.

### CODE REVIEW

File: src/Widget/Form.php [hunk 1]
```diff
@@ -104,12 +103,12 @@ public function acceptCharset(string $value): self
     }
 
     /**
-     * The action and formaction content attributes, if specified, must have a value that is a valid non-empty URL
+     * The action and formation content attributes, if specified, must have a value that is a valid non-empty URL
```
Comment (samdark) [inline src/Widget/Form.php:106]: No, it was correct.

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/FormModel.php [hunk 2]
```diff
@@ -99,18 +101,12 @@ public function getAttributePlaceholders(): array
         return [];
     }
 
-    /**
-     * @return iterable|object|scalar|Stringable|null
-     */
-    public function getAttributeCastValue(string $attribute)
+    public function getAttributeCastValue(string $attribute): iterable|int|float|string|bool|object|null
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/FormModel.php:104]: done.

File: src/Widget/Attribute/FieldAttributes.php [hunk 3]
```diff
@@ -429,20 +446,14 @@ public function readonly(bool $value = true): self
...
     {
         $new = clone $this;
-
-        if (is_string($value) || (is_object($value) && method_exists($value, '__toString'))) {
-            $new->defaultTokens[$token] = (string) $value;
-        } else {
-            throw new InvalidArgumentException('$token must be a string or \Stringable object.');
-        }
-
+        $new->defaultTokens[$token] = $value instanceof Stringable ? $value->__toString() : $value;
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Widget/Attribute/FieldAttributes.php:456]: done.

File: src/Widget/Field.php [hunk 4]
```diff
@@ -465,17 +467,17 @@ public function resetButton(array $config = [], array $attributes = []): self
      * Available methods:
      * [
      *     'encode()' => [true],
-     *     'groups()' => [['1' => ['2' => 'Moscu', '3' => 'San Petersburg']]],
-     *     'items()' => [['1' => 'Moscu', '2' => 'San Petersburg']],
+     *     'groups()' => [['1' => ['2' => 'Moscow', '3' => 'San Petersburg']]],
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Widget/Field.php:470]: done.

File: src/Widget/Form.php (hunk 1 above)
Comment (terabytesoftw) [reply src/Widget/Form.php:106]: done.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**